common/src/main/java/io/pravega/common/tracing/RPCTracingHelpers.java

```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * This interceptor is intended to get request tags from call options and attach them to the RPC request.
+     */
+    private static class FishTaggingClientInterceptor implements ClientInterceptor {
+
+        @Override
+        public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+
+            return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+                @Override
+                public void start(Listener<RespT> responseListener, Metadata headers) {
+                    // Get call options previously set by the client and attach that information to the RPC request.
+                    final String requestDescriptor = callOptions.getOption(REQUEST_DESCRIPTOR_CALL_OPTION);
+                    final String requestId = callOptions.getOption(REQUEST_ID_CALL_OPTION);
+
+                    if (requestDescriptor != null && requestId != null && !requestDescriptor.isEmpty() && !requestId.isEmpty()) {
+                        headers.put(REQUEST_DESCRIPTOR_HEADER, requestDescriptor);
+                        headers.put(REQUEST_ID_HEADER, requestId);
+                        log.info("[requestId={}] Tagging RPC request {}.", requestId, requestDescriptor);
+                    } else {
+                        log.warn("Not tagging request {}: Call options not containing request tags.", method.getFullMethodName());
+                    }
+
+                    super.start(new ForwardingClientCallListener.SimpleForwardingClientCallListener<RespT>(responseListener) {
+                        @Override
+                        public void onHeaders(Metadata headers) {
+                            super.onHeaders(headers);
+                        }
+                    }, headers);
+                }
+            };
+        }
+    }
+
+    /**
+     * This server interceptor is intended to get RPC tags from RPC headers and set the them in the request tracker.
+     */
+    private static class FishTaggingServerInterceptor implements ServerInterceptor {
+
+        @Override
+        public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, final Metadata requestHeaders, ServerCallHandler<ReqT, RespT> next) {
+            // Check if this RPC has tags to track request (e.g., older clients).
+            if (requestHeaders != null && requestHeaders.containsKey(REQUEST_DESCRIPTOR_HEADER) && requestHeaders.containsKey(REQUEST_ID_HEADER)) {
+                RequestTag requestTag = new RequestTag(requestHeaders.get(REQUEST_DESCRIPTOR_HEADER), Long.parseLong(requestHeaders.get(REQUEST_ID_HEADER)));
+                RequestTracker.getInstance().trackRequest(requestTag);
+                log.info("[requestId={}] Received tag from RPC request {}.", requestHeaders.get(REQUEST_ID_HEADER), requestHeaders.get(REQUEST_DESCRIPTOR_HEADER));
```

Comment:
This message should be debug not info. My overall recommendation is that the tracing is done in the following way:

- Existing messages should not have their severity level changed, but the trace id can be added.
- New messages related to tracing should be either trace or debug level, unless it is related to an error.

Reply:
Thanks. In all places where `requestId` appears, the log level has been preserved. Messages related to tracing are now `debug` level.